chore: aws cdk v2 migration troubleshooting (#4990)
aws-amplify/docs: src/pages/cli/migration/aws-cdk-migration.mdx

export const meta = {
  title: `AWS CDK v1 to v2 migration`,
  description: ``,
};

The version 1 of AWS Cloud Development Kit [CDK](https://docs.aws.amazon.com/cdk/v2/guide/home.html) that Amplify CLI uses under the hood has entered maintenance on June 1, 2022 with end of support date scheduled on 6/1/2023.
Amplify CLI v11 now uses version 2 of CDK to unblock Customers from deprecated CDK as well as unblock new features that won’t be added to version 1 of CDK.
To learn more about AWS CDK Migration : https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html

## When to migrate
Amplify CLI in deployment stage searches for AWS CDK v1 dependency in the amplify project and if found, prints a warning to migrate AWS CDK v2.

```
We detect you are using CDK v1 with custom stacks and overrides.AWS CDK v1 has entered
 maintenance mode on June 1, 2022 

Impacted files.
 - amplify/backend/package.json
 Upgrade '@aws-amplify/cli-extensibility-helper' to latest version ^3.0.0
 
 - amplify/backend/custom/<resource-name1>/package.json
 - amplify/backend/custom/<resource-name2>/package.json

> Follow this guide here : https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html

Reviewer: ```suggestion
Follow this guide here: https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html
```

export const meta = {
  title: `AWS CDK v1 to v2 migration`,
  description: ``,
};

AWS Cloud Development Kit [CDK version 1](https://docs.aws.amazon.com/cdk/v2/guide/home.html) that Amplify CLI uses under the hood has entered maintenance on June 1, 2022 with end of support date scheduled on 6/1/2023.
Amplify CLI v11 and above uses CDK v2 to help you move away from deprecated CDK as well as unblock new features that won’t be added to version 1 of CDK.
To learn more about AWS CDK Migration, visit [Migrating to V2](https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html)
## When to migrate
During deployment, Amplify CLI searches for AWS CDK v1 dependencies in your project and if found, Amplify CLI prints a warning to migrate to AWS CDK v2.
```

> We detect you are using CDK v1 with custom stacks and overrides.AWS CDK v1 has entered

Reviewer: ```suggestion
We detect you are using CDK v1 with custom stacks and overrides. AWS CDK v1 has entered
```

 maintenance mode on June 1, 2022 

Impacted files.
 - amplify/backend/package.json
 Upgrade '@aws-amplify/cli-extensibility-helper' to latest version ^3.0.0
 
 - amplify/backend/custom/<resource-name1>/package.json
 - amplify/backend/custom/<resource-name2>/package.json

Follow this guide here : https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html
```

## How do I migrate?

If an Amplify project using `amplify overrides [category]` or `amplify add custom` was deployed with an Amplify CLI version prior to 11.0.0, then you need to follow the steps below to migrate:
1. For projects using `amplify overrides [category]`:
   Upgrade `@aws-amplify/cli-extensibility-helper` to version `^3.0.0` in `amplify/backend/package.json`
2. For projects using `amplify add custom`:
   Update `package.json` for `custom` categories in `amplify/backend/custom/<customResourceName>/package.json` as shown below: 
   ![Amplify overrides warning](/images/ts-guide-cdk-package.png)
   And update `amplify/backend/custom/<customResourceName>/cdk-stack.ts` for `custom` categories as shown below: 
   ![Amplify custom warning](/images/ts-guide-cdk.png)

> ### Note:

Reviewer 2: we should use: `<Callout warning> </Callout>` for this. and at the top of file makes sense as its important

export const meta = {
  title: `AWS CDK v1 to v2 migration`,
  description: ``,
};

> The version 1 of AWS Cloud Development Kit [CDK](https://docs.aws.amazon.com/cdk/v2/guide/home.html) that Amplify CLI uses under the hood has entered maintenance on June 1, 2022 with end of support date scheduled on 6/1/2023.

Reviewer 3: ```suggestion
AWS Cloud Development Kit [CDK version 1](https://docs.aws.amazon.com/cdk/v2/guide/home.html) that Amplify CLI uses under the hood has entered maintenance on June 1, 2022 with end of support date scheduled on 6/1/2023.
```

> Amplify CLI v11 now uses version 2 of CDK to unblock Customers from deprecated CDK as well as unblock new features that won’t be added to version 1 of CDK.

Reviewer 3: ```suggestion
Amplify CLI v11 and above uses CDK v2 to help you move away from deprecated CDK version.
```

> To learn more about AWS CDK Migration : https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html

Reviewer 3: ```suggestion
To learn more about AWS CDK Migration, visit [Migrating to V2](https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html)
```

## When to migrate
Amplify CLI in deployment stage searches for AWS CDK v1 dependency in the amplify project and if found, prints a warning to migrate AWS CDK v2.

```
We detect you are using CDK v1 with custom stacks and overrides.AWS CDK v1 has entered
 maintenance mode on June 1, 2022 

Impacted files.
 - amplify/backend/package.json
 Upgrade '@aws-amplify/cli-extensibility-helper' to latest version ^3.0.0
 
 - amplify/backend/custom/<resource-name1>/package.json
 - amplify/backend/custom/<resource-name2>/package.json

Follow this guide here : https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html
```

## How do I migrate?

If an Amplify project using `amplify overrides [category]` or `amplify add custom` was deployed with an Amplify CLI version prior to 11.0.0, then you need to follow the steps below to migrate:

> 1. For Amplify projects using `amplify overrides [category]`:

Reviewer 3: ```suggestion
1. For projects using `amplify overrides [category]`:
```

   Upgrade `@aws-amplify/cli-extensibility-helper` to version `^3.0.0` in `amplify/backend/package.json`

> 2. For Amplify projects using `amplify add custom`:

Reviewer 3: ```suggestion
2. For projects using `amplify add custom`:
```

   Update `package.json` for `custom` categories in `amplify/backend/custom/<customResourceName>/package.json` as shown below: 
   ![Amplify overrides warning](/images/ts-guide-cdk-package.png)
   And update `amplify/backend/custom/<customResourceName>/cdk-stack.ts` for `custom` categories as shown below: 
   ![Amplify custom warning](/images/ts-guide-cdk.png)

### Note:

>  `aws-cdk-lib` version should match CDK version that ```@aws-amplify/cli-extensibility-helper``` used as dependency  in `amplify/backend/custom/<customResourceName>/package.json`.

Reviewer 3: ```suggestion
`aws-cdk-lib` version should match the CDK version that `@aws-amplify/cli-extensibility-helper` uses as a dependency  in `amplify/backend/custom/<customResourceName>/package.json`.
```

>   You can check the `aws-cdk-lib` version that `@aws-amplify/cli-extensibility-helper` uses in `package.json` [here](https://www.npmjs.com/package/@aws-amplify/cli-extensibility-helper?activeTab=explore)

Reviewer 3: ```suggestion
You can verify the `aws-cdk-lib` version that `@aws-amplify/cli-extensibility-helper` uses in the `package.json` [here](https://www.npmjs.com/package/@aws-amplify/cli-extensibility-helper?activeTab=explore)
```

export const meta = {
  title: `AWS CDK v1 to v2 migration`,
  description: ``,
};

<Callout>
AWS CDK version should match the CDK version that cli-extensibility-helper package uses as a dependency in `amplify/backend/custom/customResourceName/package.json`.

> You can verify the `aws-cdk-lib` version that `@aws-amplify/cli-extensibility-helper` uses in the `package.json`  [here](https://www.npmjs.com/package/@aws-amplify/cli-extensibility-helper?activeTab=explore).

Reviewer 4: ```suggestion
You can verify the `aws-cdk-lib` version that [`@aws-amplify/cli-extensibility-helper`](https://www.npmjs.com/package/@aws-amplify/cli-extensibility-helper?activeTab=explore) uses in the `amplify/backend/package.json` file.
```

</Callout>

> AWS Cloud Development Kit [CDK version 1](https://docs.aws.amazon.com/cdk/v2/guide/home.html) that Amplify CLI uses under the hood has entered maintenance on June 1, 2022 with end of support date scheduled on 6/1/2023.

Reviewer 4: ```suggestion
[AWS Cloud Development Kit (CDK) version 1](https://docs.aws.amazon.com/cdk/v2/guide/home.html) that Amplify CLI uses under the hood has entered maintenance on June 1, 2022 with end of support date scheduled on June 1, 2023.
```

Should the link point to the cdk v1 docs?

Reviewer 4: cc @akshbhu I've committed this change but not sure if the link should be updated

> [AWS Cloud Development Kit (CDK) version 1](https://docs.aws.amazon.com/cdk/v2/guide/home.html) that Amplify CLI uses under the hood has entered maintenance on June 1, 2022 with end of support date scheduled on June 1, 2023.

Reviewer 4: ```suggestion
[AWS Cloud Development Kit (CDK) version 1](https://docs.aws.amazon.com/cdk/v1/guide/home.html) that Amplify CLI uses under the hood has entered maintenance on June 1, 2022 with end of support date scheduled on June 1, 2023.
```

Amplify CLI v11 and above uses CDK v2 to help you move away from deprecated CDK as well as unblock new features that won’t be added to version 1 of CDK.
To learn more about AWS CDK Migration, visit [Migrating to V2](https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html)
## When to migrate
During deployment, Amplify CLI searches for AWS CDK v1 dependencies in your project and if found, Amplify CLI prints a warning to migrate to AWS CDK v2.

> ```

Reviewer 4: ```suggestion

```console
```

add language to fenced code block, separate from prose with a newline

> We detect you are using CDK v1 with custom stacks and overrides. AWS CDK v1 has entered
>  maintenance mode on June 1, 2022 

Reviewer 4: ```suggestion
We detect you are using CDK v1 with custom stacks and overrides. AWS CDK v1 has entered maintenance mode on June 1, 2022 
```

removing line break

> Impacted files.
>  - amplify/backend/package.json
>  Upgrade '@aws-amplify/cli-extensibility-helper' to latest version ^3.0.0
>  
>  - amplify/backend/custom/<resource-name1>/package.json
>  - amplify/backend/custom/<resource-name2>/package.json

Reviewer 4: ```suggestion
Impacted files:
- `amplify/backend/package.json`
	Upgrade `@aws-amplify/cli-extensibility-helper` to latest version (`^3.0.0`)
- `amplify/backend/custom/<resource-name-1>/package.json`
- `amplify/backend/custom/<resource-name-2>/package.json`
```

for readability

Follow this guide here: https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html
```

## How do I migrate?

> If an Amplify project using `amplify overrides [category]` or `amplify add custom` was deployed with an Amplify CLI version prior to 11.0.0, then you need to follow the steps below to migrate:

Reviewer 4: ```suggestion
If an Amplify project using `amplify override <category>` or `amplify add custom` was deployed with an Amplify CLI version prior to 11.0.0, then you need to follow the steps below to migrate:
```

migrate `[category]` to `<category>` to align with styleguide. Additionally in CLI help outputs the square brackets denote an optional argument/option, which can be confusing

ref https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md#16-bracket-replacement-text-in-codeblocks

> 1. For projects using `amplify overrides [category]`:

Reviewer 4: ```suggestion
1. For projects using `amplify overrides <category>`:
```

   Upgrade `@aws-amplify/cli-extensibility-helper` to version `^3.0.0` in `amplify/backend/package.json`
2. For projects using `amplify add custom`:

>    Update `package.json` for `custom` categories in `amplify/backend/custom/<customResourceName>/package.json` as shown below:

Reviewer 4: ```suggestion
   Update `package.json` for `custom` categories in `amplify/backend/custom/<custom-resource-name>/package.json` as shown below:
```

ref https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md#16-bracket-replacement-text-in-codeblocks
